components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -5384,4 +5387,180 @@ protected String getSearchQuery(String searchQuery) throws APIManagementExceptio
+
+        try {
+            if (apiIdentifier != null && type != null && name != null) {
+                requiredSequence = getDefaultSequence(type, name);
+                if (log.isDebugEnabled()) {
+                    log.debug("Get the default sequences.");
+                }
+                if (requiredSequence == null) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Get the custom sequences.");
+                    }
+                    requiredSequence = getCustomSequence(apiIdentifier, type, name);
+                }
+
+                //Convert the content stream to a string.
+                if (requiredSequence != null) {
+                    sequenceStream = requiredSequence.getContentStream();
+                    StringWriter stringWriter = new StringWriter();
+                    IOUtils.copy(sequenceStream, stringWriter);
+                    sequenceText = stringWriter.toString();
+                } else {
+                    log.error("No sequence for the name " + name + "is found!");
+                }
+            } else {
+                log.error("Invalid arguments.");
+            }
+        }  catch (APIManagementException e) {
+            log.error(e.getMessage());
+            throw new APIManagementException(e);
+        } catch (RegistryException e) {
+            log.error(e.getMessage());
+            throw new APIManagementException(e);
+        } catch (IOException e) {
+            log.error(e.getMessage());
+            throw new APIManagementException(e);
+        }
+        return sequenceText;
+    }
+
+    /**
+     * Get the mediation sequence which matches the given type and name from the custom sequences.
+     * @param type : The sequence type.
+     * @param name : The name of the sequence.
+     * @return : The mediation sequence which matches the given parameters. Returns null if no matching sequence is
+     * found.
+     */
+    private Resource getDefaultSequence(String type, String name) throws APIManagementException {
+        String defaultSequenceFileLocation = "";
+
+        try {
+            UserRegistry registry = ServiceReferenceHolder.getInstance().getRegistryService()
+                    .getGovernanceSystemRegistry(tenantId);
+
+            if ("fault".equals(type)) {
+                defaultSequenceFileLocation = APIConstants.API_CUSTOM_FAULTSEQUENCE_LOCATION;
+            } else if ("out".equals(type)) {
+                defaultSequenceFileLocation = APIConstants.API_CUSTOM_OUTSEQUENCE_LOCATION;
+            } else {
+                defaultSequenceFileLocation = APIConstants.API_CUSTOM_INSEQUENCE_LOCATION;
+            }
+            if (registry.resourceExists(defaultSequenceFileLocation)) {
+                org.wso2.carbon.registry.api.Collection defaultSeqCollection =
+                        (org.wso2.carbon.registry.api.Collection) registry.get(defaultSequenceFileLocation);
+                if (defaultSeqCollection !=null) {
```

Comment:
Formatting issue

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/5114/commits/c6bc99c70b7ef3df0181e7565b77152ab729b066